### PR TITLE
Support for storage variables

### PR DESCRIPTION
**Pull request by matharoo**
## what
* [x] Add support for storage variables `storage_type`, `iops` and `allocated_storage`.
* [x] Fix broken tests

## why
* Be able to specify the size and type of storage of the database in the cluster.

## references
* closes https://github.com/cloudposse/terraform-aws-rds-cluster/issues/140
* [`aws_rds_cluster`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/rds_cluster)

### CODE REVIEW

File: variables.tf [hunk 1]
```diff
@@ -204,6 +204,24 @@ variable "storage_encrypted" {
...
+variable "storage_type" {
+  type        = string
+  description = "One of 'standard' (magnetic), 'gp2' (general purpose SSD), or 'io1' (provisioned IOPS SSD)"
+  default     = "standard"
+}
+
+variable "iops" {
+  type        = number
+  description = "The amount of provisioned IOPS. Setting this implies a storage_type of 'io1'. Default is 0 if rds storage type is not 'io1'"
+  default     = 0
```
Comment (nitrocode) [inline variables.tf:216]: Can we default this to null?

```suggestion
  default     = null
```

File: variables.tf [hunk 2]
```diff
@@ -204,6 +204,24 @@ variable "storage_encrypted" {
   default     = false
 }
 
+variable "storage_type" {
+  type        = string
+  description = "One of 'standard' (magnetic), 'gp2' (general purpose SSD), or 'io1' (provisioned IOPS SSD)"
+  default     = "standard"
```
Comment (nitrocode) [inline variables.tf:210]: The docs do not mention standard nor gp2 and say the default is io1.

Are the docs out of date? Can we at least default this to null or io1 as it says? We dont want to break existing configs for anyone.

```suggestion
  default     = "io1"
```

Comment (matharoo) [reply variables.tf]: yes the tf docs say the default is `io1`. I have updated this.

File: variables.tf (hunk 1 above)
Comment (matharoo) [reply variables.tf:216]: I set this based on what we have in 
[terraform-aws-rds](https://github.com/cloudposse/terraform-aws-rds) module @ https://github.com/cloudposse/terraform-aws-rds/blob/558eb8f2aac9dc79b741f866bbbf8bb44d73e911/variables.tf#L80 

According to TF docs for `aws_rds_cluster`. This is an optional var, we can set this to `null`. 

On a side note though TF docs dont say any default value for this but its needed for Multi a-z cluster,  then this value ranges from min `1000` to number based on db engine, https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/CHAP_Storage.html#USER_PIOPS

**Review by nitrocode — DISMISSED** [review]

**Review by nitrocode — COMMENTED** [review]
Please fix tests and bump versions.tf to use aws 4.x provider

**Review by nitrocode — COMMENTED** [review]
Please update any modules in the examples as some of them may point to aws 3.x provider

**Comment by matharoo** [comment]
```
# Error: Unsupported block type
# 
#   on .terraform/modules/subnets/moved.tf line 3:
#    3: moved {
# 
# Blocks of type "moved" are not expected here.
```
The tests now giving an error about the `moved` block in newer version of subnet module. The `moved` [block](https://nedinthecloud.com/2021/12/14/using-the-moved-block-in-terraform-1-1/) was introduced in terraform `1.1`   . I will update the test example terraform version to fix this error.

**Comment by matharoo** [comment]
I found the terratest was older version `v0.16.0`, the double quotes issue was reported on terratest [github](https://github.com/gruntwork-io/terratest/issues/717) as well. It was [fixed](https://github.com/gruntwork-io/terratest/releases/tag/v0.31.0) in v0.31.0 . I have now upgraded the version of terratest to `v0.40.15`

**Comment by matharoo** [comment]
@nitrocode terratest passes now, however  the failing `test/bats` can be ignored as mentioned in another PR [here](https://github.com/cloudposse/terraform-aws-rds-cluster/pull/138#pullrequestreview-1003797976)

**Comment by nitrocode** [comment]
cc: @Nuru test/bats are failing. Is this OK to merge as part of the terraform module 1.0.0 release ?

**Review by nitrocode — APPROVED** [review]

**Pull request merged**